java/client/src/main/java/glide/api/commands/GenericCommands.java

```diff
@@ -132,4 +134,90 @@ CompletableFuture<Boolean> copy(
      * }</pre>
      */
     CompletableFuture<String> randomKey();
+
+    /**
+     * Sorts the elements in the list, set, or sorted set at <code>key</code> and returns the result.
+     * The <code>sort</code> command can be used to sort elements based on different criteria and
+     * apply transformations on sorted elements.<br>
+     * To store the result into a new key, see {@link #sortStore(String, String, SortOptions)}.
+     *
+     * @param key The key of the list, set, or sorted set to be sorted.
+     * @param sortOptions The {@link SortOptions}.
+     * @return A <code>Array</code> of sorted elements.
+     * @example
+     *     <pre>{@code
+     * client.hset("user:1", Map.of("name", "Alice", "age", "30")).get();
+     * client.hset("user:2", Map.of("name", "Bob", "age", "25")).get();
+     * client.lpush("user_ids", new String[] {"2", "1"}).get();
+     * String [] payload = client.sort(
+     *      "user_ids",
+     *      SortOptions
+     *          .builder()
+     *              .byPattern("user:*->age")
+     *                  .getPatterns(new String[] {"user:*->name"})
+     *                      .build()).get();
```

Comment:
You can collapse code in docs into 1 or 2 lines - for spotless it is just a comment. Just keep it readable.